vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/AbstractGridMultiSelectionModel.java

```diff
@@ -375,7 +375,8 @@ void clientSelectAll() {
             allItemsStream = fetchAllHierarchical(
                     (HierarchicalDataProvider<T, ?>) dataProvider);
         } else {
-            allItemsStream = dataProvider.fetch(new Query<>());
+            allItemsStream = dataProvider.fetch(getGrid().getDataCommunicator()
```

Comment:
My initial thought is that this should also be applied in `AbstractGridMultiSelectionModel.selectAll` / when using server-side selection.

Reply:
This is true. Addressed in the new commit.